bigquery/google/cloud/bigquery/job.py

```diff
@@ -742,7 +742,10 @@ def from_api_repr(cls, resource, client):
         dataset = Dataset(dest_config['datasetId'], client)
         destination = Table(dest_config['tableId'], dataset)
         sources = []
-        for source_config in config['sourceTables']:
+        source_configs = config.get('sourceTables')
+        if source_configs is None:
+            source_configs = [config['sourceTable']]
```

Comment:
That is to say, why don't we throw our own error rather than an opaque `KeyError`, with an explanation of why our interpretation of the docs say the server should never return values in that state